desc/__init__.py

```diff
@@ -101,10 +102,162 @@ def set_device(kind="cpu"):
             if mem > maxmem:
                 maxmem = mem
                 selected_gpu = dev
-        config["device"] = selected_gpu["type"] + " (id={})".format(
-            selected_gpu["index"]
-        )
+        config["device"] = selected_gpu["type"] + " (id=%s)" % selected_gpu["index"]
         config["avail_mem"] = (
             selected_gpu["mem_total"] - selected_gpu["mem_used"]
         ) / 1024  # in GB
         os.environ["CUDA_VISIBLE_DEVICES"] = str(selected_gpu["index"])
+
+
+logging.getLogger().addHandler(logging.NullHandler())
+# DESC automatically configures it's own logger, not the root logger, so we give the
+# root logger a NullHandler.  DESC's logger config can be changed with the following
+# functions, or with flags passed to any input-reading desc function as desired, with
+# the 'quiet' (-q), or 'disable_logging' (-d) flags to prevent messages propogating
+# into your own logging solution.  'disable logging' will toggle logfile logging off,
+# and the 'quiet' command will turn off console logging.
+
+
+def stop_logfile_logging():
+    """Quickly stops logfile logging to DESC logger.
+
+    Returns
+    -------
+        bool :
+            Returns True if logfile logger was successfully turned off.
+    """
+    logger = logging.getLogger("DESC_logger")
+    if logger.handlers is not None:
+        for handler in logger.handlers:
+            if isinstance(handler, logging.handlers.RotatingFileHandler):
+                handler.setLevel("CRITICAL")
+                return True
+            else:
+                warnings.warn(
+                    colored("No DESC logfile logger has been found.", "yellow")
+                )
+    else:
+        warnings.warn(colored("No DESC logger has been found.", "yellow"))
+        return False
+
+
+def set_logfile_logging(logfile_level="DEBUG", logfile_file="desc.log"):
```

Comment:
does "logfile handler" just mean this will tell the logger to send logging statements of a certain verbosity to the specified logfile?